lib/src/front_end/piece_writer.dart

```diff
@@ -52,23 +57,186 @@ class PieceWriter {
   /// This can only be accessed if there is a selection.
   late final int _selectionEnd = _findSelectionEnd();
 
+  /// The stack of pieces being built by calls to [build()].
+  ///
+  /// Each call to [build()] pushes a new list onto this stack. All of the
+  /// pieces written during that call to [build()] end up in that list and are
+  /// returned by as an [AdjacentPiece] (or just the single piece if there is
+  /// only one).
+  final List<List<Piece>> _pieces = [];
+
+  /// The most recent piece, if it's a [CodePiece] that can have more code
```

Comment:
Can you elaborate on "Most recent"? I think it means the _last_ element in the last list in `_pieces`?